bindings/java/c/host.c

```diff
@@ -0,0 +1,691 @@
+    }
+    return result;
+}
+
+size_t get_code_size_fn(struct evmc_host_context* context,const evmc_address* address){
+    size_t result = 0;
+    char java_method_name[] = "get_code_size";
+    char java_method_signature[] = "(I[B)I";
+    JNIEnv *jenv = attach();
+    if(jenv != NULL) {
+        jclass host_class;
+        jmethodID method;
+        jint jcontext_index;
+        jbyteArray jaddress;
+
+        //get java class
+        host_class = (*jenv)->FindClass(jenv, "org/ethereum/evmc/Host");
+        if(!host_class){
+            printf("JNI Error: FindClass was unable to load the Host class\n");
+            detach(jenv);
+        }
+    
+        //get java method
+        method = (*jenv)->GetStaticMethodID(jenv, host_class, java_method_name, java_method_signature);
+        if(!method){
+            printf("JNI Error: GetStaticMethodID was unable to find method %s\n",java_method_name);
+            detach(jenv);
+        }
+
+        //set java method params
+
+        if (!context) {
+            printf("JNI Error: context is NULL in method %s\n",java_method_name);
+            detach(jenv);
+        }
+        jcontext_index = context->index;
+
+        jaddress = (*jenv)->NewByteArray(jenv, ADDRESS_SIZE);
+        (*jenv)->SetByteArrayRegion(jenv, jaddress, 0, ADDRESS_SIZE, (jbyte *)address);
+        
+        if(!jaddress){
+            printf("JNI Error: method params could not be initialized for method: %s\n",java_method_name);
+            detach(jenv);
+        }
+
+        //call java method
+        jint jresult = (*jenv)->CallStaticIntMethod(jenv, host_class, method, jcontext_index, jaddress);
+        result = jresult;
+    }
+    return result;
+}
+
+evmc_bytes32 get_code_hash_fn(struct evmc_host_context* context, const evmc_address* address){
+    evmc_bytes32 result;
+    char java_method_name[] = "get_code_hash";
+    char java_method_signature[] = "(I[B)Ljava/nio/ByteBuffer;";
+    JNIEnv *jenv = attach();
+    if(jenv != NULL) {
+        jclass host_class;
+        jmethodID method;
+        jint jcontext_index;
+        jbyteArray jaddress;
+
+        //get java class
+        host_class = (*jenv)->FindClass(jenv, "org/ethereum/evmc/Host");
+        if(!host_class){
+            printf("JNI Error: FindClass was unable to load the Host class\n");
+            detach(jenv);
+        }
+    
+        //get java method
+        method = (*jenv)->GetStaticMethodID(jenv, host_class, java_method_name, java_method_signature);
+        if(!method){
+            printf("JNI Error: GetStaticMethodID was unable to find method %s\n",java_method_name);
+            detach(jenv);
+        }
+
+        //set java method params
+
+        if (!context) {
+            printf("JNI Error: context is NULL in method %s\n",java_method_name);
+            detach(jenv);
+        }
+        jcontext_index = context->index;
+
+        jaddress = (*jenv)->NewByteArray(jenv, ADDRESS_SIZE);
```

Comment:
Instead of `ADDRESS_SIZE` I'd suggest to use `sizeof(struct evmc_address)`. Here and other appropriate places.